src/ResourceManager/PowerBI/Commands.PowerBI/Commands/SuspendAzureRmPowerBIEmbeddedCapacity.cs

```diff
@@ -0,0 +1,110 @@
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using Microsoft.Azure.Commands.PowerBI.Models;
+using Microsoft.Azure.Management.PowerBIDedicated;
+using System.Collections.Generic;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.PowerBI.Properties;
+using Microsoft.Azure.Commands.PowerBI.Utilities;
+using Microsoft.Azure.Management.PowerBIDedicated.Models;
+
+namespace Microsoft.Azure.Commands.PowerBI
+{
+    [Cmdlet(VerbsLifecycle.Suspend, "AzureRmPowerBIEmbeddedCapacity", SupportsShouldProcess = true),
+        OutputType(typeof(PSDedicatedCapacity))]
+    public class SuspendAzurePowerBIEmbeddedCapacity : PowerBICmdletBase
+    {
+        protected const string CmdletParametersSet = "ByNameAndResourceGroup";
+        protected const string ObjectParameterSet = "ByInputObject";
+        protected const string ResourceIdParameterSet = "ByResourceId";
+
+        [Parameter(
+            ParameterSetName = CmdletParametersSet,
+            Mandatory = true,
+            Position = 0,
+            HelpMessage = "Name of a specific capacity.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            ParameterSetName = CmdletParametersSet,
+            Mandatory = false,
+            HelpMessage = "Name of resource group under which to retrieve the capacity.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            ParameterSetName = ResourceIdParameterSet,
+            Mandatory = true,
+            Position = 0,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "PowerBI Embedded Capacity ResourceID.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(
+            ParameterSetName = ObjectParameterSet,
+            Mandatory = true,
+            Position = 0,
+            ValueFromPipeline = true,
+            HelpMessage = "PowerBI Embedded Capacity object.")]
+        [ValidateNotNullOrEmpty]
+        public PSDedicatedCapacity InputObject { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public SwitchParameter PassThru { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            string capacityName = string.Empty;
```

Comment:
@selasagi same comment about assigning this to `Name` and removing the if-statement